File: symfony/src/Entity/SpecificScore.php
<?php

namespace App\Entity;

use App\Repository\SpecificScoreRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: SpecificScoreRepository::class)]
#[ORM\Table(name: 'specific_score')]
class SpecificScore
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    private ?int $userId = null;

    #[ORM\Column]
    private ?int $specificTestId = null;

    #[ORM\Column]
    private ?int $totalScore = 0;

    #[ORM\Column]
    private ?int $maxScore = 0;

    #[ORM\Column]
    private ?int $percentage = 0;

    #[ORM\Column(length: 100)]
    private ?string $category = null;

    #[ORM\Column(length: 20)]
    private ?string $level = 'Faible';

    #[ORM\Column]
    private ?int $weekNumber = null;

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $passedAt = null;

    public function __construct()
    {
        $this->passedAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getUserId(): ?int { return $this->userId; }
    public function setUserId(int $userId): static { $this->userId = $userId; return $this; }
    public function getSpecificTestId(): ?int { return $this->specificTestId; }
    public function setSpecificTestId(int $specificTestId): static { $this->specificTestId = $specificTestId; return $this; }
    public function getTotalScore(): ?int { return $this->totalScore; }
    public function setTotalScore(int $totalScore): static { $this->totalScore = $totalScore; return $this; }
    public function getMaxScore(): ?int { return $this->maxScore; }
    public function setMaxScore(int $maxScore): static { $this->maxScore = $maxScore; return $this; }
    public function getPercentage(): ?int { return $this->percentage; }
    public function setPercentage(int $percentage): static { $this->percentage = $percentage; return $this; }
    public function getCategory(): ?string { return $this->category; }
    public function setCategory(string $category): static { $this->category = $category; return $this; }
    public function getLevel(): ?string { return $this->level; }
    public function setLevel(string $level): static { $this->level = $level; return $this; }
    public function getWeekNumber(): ?int { return $this->weekNumber; }
    public function setWeekNumber(int $weekNumber): static { $this->weekNumber = $weekNumber; return $this; }
    public function getPassedAt(): ?\DateTimeInterface { return $this->passedAt; }
    public function setPassedAt(\DateTimeInterface $passedAt): static { $this->passedAt = $passedAt; return $this; }
}

File: integrated-project/mindboost-javafx/src/main/java/com/mindboost/app/model/SpecificScore.java
package com.mindboost.app.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "specific_score")
public class SpecificScore {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "specific_test_id", nullable = false)
    private Integer specificTestId;

    @Column(name = "total_score", nullable = false)
    private Integer totalScore = 0;

    @Column(name = "max_score", nullable = false)
    private Integer maxScore = 0;

    @Column(nullable = false)
    private Integer percentage = 0;

    @Column(length = 100, nullable = false)
    private String category;

    @Column(length = 20, nullable = false)
    private String level = "Faible";

    @Column(name = "week_number", nullable = false)
    private Integer weekNumber = 1;

    @Column(name = "passed_at", nullable = false)
    private LocalDateTime passedAt = LocalDateTime.now();

    public Integer getId() {
        return id;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public Integer getSpecificTestId() {
        return specificTestId;
    }

    public void setSpecificTestId(Integer specificTestId) {
        this.specificTestId = specificTestId;
    }

    public Integer getTotalScore() {
        return totalScore;
    }

    public void setTotalScore(Integer totalScore) {
        this.totalScore = totalScore;
    }

    public Integer getMaxScore() {
        return maxScore;
    }

    public void setMaxScore(Integer maxScore) {
        this.maxScore = maxScore;
    }

    public Integer getPercentage() {
        return percentage;
    }

    public void setPercentage(Integer percentage) {
        this.percentage = percentage;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    public Integer getWeekNumber() {
        return weekNumber;
    }

    public void setWeekNumber(Integer weekNumber) {
        this.weekNumber = weekNumber;
    }

    public LocalDateTime getPassedAt() {
        return passedAt;
    }

    public void setPassedAt(LocalDateTime passedAt) {
        this.passedAt = passedAt;
    }
}
